File: Server/src/main/java/ltd/boxin/config/ConfigCopyAndCreate.java
package ltd.boxin.config;

import com.google.gson.Gson;
import com.google.gson.internal.LinkedTreeMap;

import java.io.*;
import java.util.Objects;

/**
 * @author baka4n
 * 配置文件为空时候默认从资源列表抽出
 */
public class ConfigCopyAndCreate {
    public static final Gson gson = new Gson();
    public static int port = 25598;
    public static String host = "127.0.0.1";

    public static void copy() throws IOException {
        File file = new File(System.getProperty("user.dir"), ".server/config.json");
        if (!file.getParentFile().exists()) {
            file.getParentFile().mkdirs();
        }
        InputStream in = null;
        OutputStream out = null;
        try {
            in = ConfigCopyAndCreate.class.getResourceAsStream("/config.json");
            out = new FileOutputStream(file);
            byte[] buf = new byte[1024];
            int bytesRead;
            while ((bytesRead = in.read(buf)) > 0) {
                out.write(buf, 0, bytesRead);
            }
        } catch (IOException e) {
            System.out.println(e.getMessage());
        } finally {
            in.close();
            out.close();
        }

    }

    public static void init() {
        try {
            copy();
        } catch (IOException e) {
            System.out.println(e.getMessage());
        }
        var linkedTreeMap = gson.fromJson(new InputStreamReader(Objects.requireNonNull(ConfigCopyAndCreate.class.getResourceAsStream("/config.json"))), LinkedTreeMap.class);
        if (linkedTreeMap.get("port") instanceof Double d) {
            port = (int) d.doubleValue();
        }
    }
}

File: src/main/kotlin/api/ConfigCopyAndCreate.kt
package api

import com.google.gson.Gson
import com.google.gson.internal.LinkedTreeMap
import java.io.*

object ConfigCopyAndCreate {
   val gson = Gson()
   var port = 25599
   var host = "127.0.0.1"

   @Throws(IOException::class)
   fun copy() {
      val file = File(System.getProperty("user.dir"), ".client/config.json")
      if (!file.parentFile.exists()) {
         file.parentFile.mkdirs()
      }
      var `in`: InputStream? = null
      var out: OutputStream? = null
      try {
         `in` = javaClass.getResourceAsStream("/config.json")
         out = FileOutputStream(file)
         val buf = ByteArray(1024)
         var bytesRead: Int
         while (`in`.read(buf).also { bytesRead = it } > 0) {
            out.write(buf, 0, bytesRead)
         }
      } catch (e: IOException) {
         println(e.message)
      } finally {
         `in`!!.close()
         out!!.close()
      }
   }

   fun init() {
      try {
         copy()
      } catch (e: IOException) {
         println(e.message)
      }
      var linkedTreeMap = gson.fromJson(InputStreamReader(javaClass.getResourceAsStream("/config.json")!!), LinkedTreeMap::class.java)
      val get = linkedTreeMap.get("port")
      if (get is Double) {
         port = get.toInt()
      }
      val host = linkedTreeMap.get("ip")
      if (host is String?) {
         this.host = host!!
      }


   }
}